capstone/src/main/resources/static/script.js

```diff
@@ -83,40 +82,74 @@ function checkAuth(){
   if (email == "") {
       email = "jennysheng@google.com";
   }
-  // prefill the email in the form so that user cannot edit their email
+  return email;
+}
+
+/* get the user information for the profile page */
+function getUser() {
+  var email = checkAuth();
+  
+   // prefill the email in the form so that user cannot edit their email
   document.getElementById("email-form-display").innerText = email;
   document.getElementById("org-email-form-display").innerText = email;
   document.getElementById("email-form").value = email;
   document.getElementById("org-email-form").value = email;
-  return email;
+
+  var userType = sessionStorage.getItem("user-type");
+
+  // if there is no userType stored in this session, that means this is a new user
+  if (userType == null) {
+    getUserType();
+    displayForm("individual", true);
+  } else {
+    if (userType == "organization") {
+      fetch('get-organization?email=' + email).then(response => response.json()).then((data) => {
+        createProfile(data, true);
+        displayForm(data[0].userType, false);
+        sessionStorage.setItem("user-type", data[0].userType);
+      });
+    } else if (userType == "individual") {
+      fetch('get-individual?email=' + email).then(response => response.json()).then((data) => {
+        createProfile(data, false);
+        displayForm(data[0].userType, false);
+        document.getElementById("profile-section").style.display = "block";
+        document.getElementById("no-profile").style.display = "none";
+        sessionStorage.setItem("user-type", data[0].userType);
+      });  
+    }
+    // display profile
+    document.getElementById("profile-section").style.display = "block";
+    document.getElementById("no-profile").style.display = "none";
+  }
 }
 
-/* gets the user information from Datastore and display them in profile */
-function getUser() {
+/* get and store the user type (individual or organization). With this we determine whether
+this user exists in our database or not */
+function getUserType() {
     var email = checkAuth();
 
     /* Since there is no way to know beforehand whether the user is an organization 
     or an individual, we have to do two fetches to check the organization entities and
     the user entities */
     fetch('get-organization?email=' + email).then(response => response.json()).then((data) => {
```

Comment:
Out of scope for this PR: consider introducing a new server-side endpoint `get-user-type`; then you won't need to sequential calls.

Reply:
How would this look with the two separate controllers that I have? If I have this in one controller the other user type can't be searched. Or would this reside in a separate controller?

Reply:
Separate controller, I think. On the backend it will still have to check both people and organizations, but at least your front-end will have a clean abstraction.

// non-actionable aside
In database design, this kind of problem where some fields are shared and some are not could be solved with 3 tables: "user", "person-user" and "organization-user". The "user" table would have a column for type, and the other two tables would contain fields not relevant to the other type.
As a completely different solution, you could solve this on the front-end by having two completely distinct screens for people and organizations. 
So, there are many ways to approach the same usecase.

Reply:
Filed this as an issue to be addressed in a later PR.